File: ClientApp/src/app/models/Article.ts
import { Product } from "./Product";
import { User } from "./User";

export class Article{    
    public idArticle?: number;
    public rating: number;
    public text: string;
    public time: Date;
    public product?: Product;
    public user?: User;
    public comments?: Comment[];

    constructor(params: Article = {} as Article){
        let{
            idArticle = undefined,
            rating = undefined,
            text = undefined,
            time = undefined,
            comments = [],
            product = undefined,
            user = undefined
        } = params;
        this.idArticle = idArticle;
        this.rating = rating;
        this.text = text;
        this.time = time;
        this.comments = comments;
        this.product = product;
        this.user = user;
    }
}

File: Models/Data/Article.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace ProductsReviewsAngular.Models
{
    public class Article
    {
        public int idArticle { get; set; }
        public int rating { get; set; }
        public string text { get; set; }
        public virtual Product product { get; set; }
        public virtual User user { get; set; }
        public DateTime time {get;set;}
        public virtual ICollection<Comment> comments { get; set; }

        public override string ToString()
        {
            return "id: " + idArticle + " rating: " + rating + " text: " + text + " Product: " + product + " User: " + user + " time: " + time;
        }
    }
}
